users/user-guide-query/multi-stage-query/operator-types/aggregate.md

```diff
@@ -0,0 +1,194 @@
+processed in the _coordinator_ stage.
+
+In Apache Pinot 1.1.0, the multi-stage query engine always keeps the data in memory.
+This means that the amount of memory used by the engine is proportional to the number of groups generated by the
+`group by` clause and the amount of data that needs to be kept for each group (which depends on the aggregation 
+function).
+
+Even when the aggregation function is a simple `count`, which only requires to keep a long for each group in memory,
+the amount of memory used can be high if the number of groups is high.
+This is why the engine limits the number of groups.
+By default, this limit is 100.000, but this can be changed by providing hints.
+
+### Blocking nature
+The aggregate operator is a blocking operator. It needs to consume all the input data before emitting the result.
+
+## Hints
+
+### num_groups_limit
+Type: Integer
+
+Default: 100.000
+
+Defines the max number of groups that can be created by the `group by` clause. 
+If the number of groups exceeds this limit, the query will not fail but will stop the execution.
+  
+Example:
+```sql
+SELECT
+/*+  aggOptions(num_groups_limit='10000000') */
+    col1, count(*)
+FROM table GROUP BY col1
+```
+
+### is_partitioned_by_group_by_keys
+Type: Boolean 
+
+Default: false
+
+If set to true, the engine will consider that the data is already partitioned by the `group by` keys.
+This means that the engine will not need to shuffle the data to group them by the `group by` keys and the
+_coordinator_ stage will be able to compute the final result without needing to merge the partial results.
+
+{% hint style="warning" %}
+Caution: This hint should only be used if the data is already partitioned by the `group by` keys. 
+There is no check to verify that the data is indeed partitioned by the `group by` keys and using this hint when the data
+is not partitioned by the `group by` keys will lead to incorrect results.
+{% endhint %}
+
+Example:
+```sql
+SELECT 
+/*+ aggOptions(is_partitioned_by_group_by_keys='true') */
+    a.col3, a.col1, SUM(b.col3)
+FROM a JOIN b ON a.col3 = b.col3 
+GROUP BY a.col3, a.col1
+```
+
+### is_skip_leaf_stage_group_by
+Type: Boolean
+
+Default: false
+
+If set to true, the engine will not push down the aggregate into the leaf stage.
+In some situations, it could be wasted effort to do group-by on leaf, eg: when cardinality of group by column is very 
+high.
+
+Example:
+```sql
+SELECT 
+/*+ aggOptions(is_skip_leaf_stage_group_by='true') */ 
+    a.col1, SUM(a.col3) 
+FROM a
+WHERE a.col3 >= 0 AND a.col2 = 'a' 
+GROUP BY a.col1
+```
+
+### max_initial_result_holder_capacity
+Type: Integer
+
+Default: 10.000
+
+Defines the initial capacity of the result holder that stores the intermediate results of the aggregation.
+This hint can be used to reduce the memory usage of the engine by setting a value close to the expected number of 
```

Comment:
this is for server? or coordinator? or both?

Reply:
This is in fact a concept from single-stage that we replicate in multi-stage intermediary stages. I'm not 100% sure if changing the hint also modifies the group bys executed in the leaf stage operator.

This for sure doesn't affect the broker, given that it doesn't execute the final reduce phase in multi-stage